packages/phone-number-privacy/signer/src/signer/sign.abstract.ts

```diff
@@ -0,0 +1,62 @@
+import {
+  DomainRestrictedSignatureRequest,
+  KEY_VERSION_HEADER,
+  SignMessageRequest,
+  WarningMessage,
+} from '@celo/phone-number-privacy-common'
+import Logger from 'bunyan'
+import { Request } from 'express'
+import { computeBlindedSignature } from '../bls/bls-cryptography-client'
+import { getKeyProvider } from '../key-management/key-provider'
+import { Key } from '../key-management/key-provider-base'
+import { IAction, Session } from './action.interface'
+import { IOAbstract } from './io.abstract'
+
+declare type OdisSignatureRequest = SignMessageRequest | DomainRestrictedSignatureRequest
+
+export abstract class SignAbstract<R extends OdisSignatureRequest> implements IAction<R> {
+  abstract readonly io: IOAbstract<R>
+
+  public abstract perform(session: Session<R>): Promise<void>
+
+  protected async sign(
```

Comment:
Having this defined in an abtract class may not be that valuable because each of the two concrete types will need to use the cooresponsing OPRF or POPRF functions in the concrete implementation.

Reply:
Fair point, I'm happy to cross that bridge when we get to it. My intuition is that it won't hurt to have this since it allows us to factor out common code like logging and keystore fetching into the same interface. Maybe when the POPRF function is ready we can inject it into this class